src/main/java/com/couchbase/client/dcp/config/ClientEnvironment.java

```diff
@@ -156,6 +175,16 @@ private ClientEnvironment(final Builder builder) {
         socketConnectTimeout = builder.socketConnectTimeout;
         dcpChannelsReconnectDelay = builder.dcpChannelsReconnectDelay;
         dcpChannelsReconnectMaxAttempts = builder.dcpChannelsReconnectMaxAttempts;
+        if (builder.eventBus != null) {
+            eventBus = builder.eventBus;
+            this.scheduler = null;
+            this.schedulerShutdownHook = new NoOpShutdownHook();
+        } else {
+            CoreScheduler scheduler = new CoreScheduler(3);
```

Comment:
I wonder if we should make this part of the environment in general and make it overridable